kernel/kernel-api/src/main/java/io/delta/kernel/utils/CloseableIterator.java

```diff
@@ -34,6 +37,39 @@
  */
 @Evolving
 public interface CloseableIterator<T> extends Iterator<T>, Closeable {
+
+    /**
+     * Returns true if the iteration has more elements. (In other words, returns true if next would
+     * return an element rather than throwing an exception.)
+     *
+     * @return true if the iteration has more elements
+     * @throws KernelEngineException For any underlying exception occurs in {@link Engine} while
+     *                               trying to execute the operation. The original exception is (if
+     *                               any) wrapped in this exception as cause. E.g.
+     *                               {@link IOException} thrown while trying to read from a Delta
+     *                               log file. It will be wrapped in this exception as cause.
+     * @throws KernelException       When encountered an operation or state that is invalid or
```

Comment:
for `KernelEngineException` you state i'ts for exceptions that occur in the Engine.

Am I correct in assuming that `KernelException` is for exceptions that occur in the Kernel, not in the engine? Can you add that very short clarification here?

Reply:
That is correct. Clarified.